nodeadm/internal/featuregates/featuregates.go

```diff
@@ -0,0 +1,35 @@
+package featuregates
+
+type Feature string
+
+const (
+	InstanceIdNodeName Feature = "InstanceIdNodeName"
+)
+
+type FeatureGate struct {
+	// Key of the feature
+	Key Feature
+	// Verifier that will validate input and set default value
+	Verifier func(Feature, map[Feature]bool) bool
+}
+
+func DefaultTrue(key Feature, featureGates map[Feature]bool) bool {
+	enabled, set := featureGates[key]
+	return !set || enabled
+}
+
+func DefaultFalse(key Feature, featureGates map[Feature]bool) bool {
+	enabled, set := featureGates[key]
+	return set && enabled
+}
+
+func (fg FeatureGate) IsEnabled(featureGates map[Feature]bool) bool {
+	return fg.Verifier(fg.Key, featureGates)
+}
+
+// InstanceIdNodeNameGate controls whether to use instance ID as node name for AL2023 node group.
+// By default, this feature is disabled, and the private DNS Name will be used.
```

Comment:
```suggestion
// InstanceIdNodeNameGate controls whether to use instance ID as the node's name.
// By default, this feature is disabled, and the private DNS Name will be used.
```

I think the distro specific details are unrelated to the feature gate